pkg/machines/components/vm/memoryModal.jsx

```diff
@@ -0,0 +1,256 @@
+        let maxMemory = convertToUnit(this.state.maxMemory, this.state.maxMemoryUnit, 'MiB');
+        memory = convertToUnit(memory, this.state.memoryUnit, 'MiB');
+
+        memory = clamp(memory, maxMemory, 1);
+
+        memory = convertToUnit(memory, 'MiB', this.state.memoryUnit);
+
+        this.setState({ memory: memory });
+    }
+
+    onMaxMemoryBlur (maxMemory) {
+        let memory = convertToUnit(this.state.memory, this.state.memoryUnit, 'MiB');
+        maxMemory = convertToUnit(maxMemory, this.state.maxMemoryUnit, 'MiB');
+
+        if (maxMemory < memory)
+            maxMemory = memory;
+
+        maxMemory = convertToUnit(maxMemory, 'MiB', this.state.maxMemoryUnit);
+
+        this.setState({ maxMemory: maxMemory });
+    }
+
+    onValueChanged(key, value) {
+        const stateDelta = { [key]: value };
+
+        this.setState(stateDelta);
+    }
+
+    onUnitsValueChanged (key, unit) {
+        const memory = Math.round(convertToUnit(this.state[key], this.state[key + "Unit"], unit));
+
+        this.setState({ [key]: memory, [key + "Unit"]: unit });
+    }
+
+    dialogErrorSet(text) {
+        this.setState({ dialogError: text });
+    }
+
+    dialogErrorDismiss() {
+        this.setState({ dialogError: undefined });
+    }
+
+    close() {
+        this.setState({ showModal: false });
+        this.dialogErrorDismiss();
+    }
+
+    open() {
+        this.setState({ showModal: true });
+    }
+
+    save() {
+        const { dispatch, vm } = this.props;
+
+        if (this.state.maxMemory !== vm.memory || this.state.memory !== vm.currentMemory) {
+            const maxMemory = parseInt(convertToUnit(this.state.maxMemory, this.state.maxMemoryUnit, 'KiB'));
+            const memory = parseInt(convertToUnit(this.state.memory, this.state.memoryUnit, 'KiB'));
+
+            // We want to change offline xml wherever machine is running or not
+            dispatch(setMemorySettings(
+                vm,
+                maxMemory,
+                memory
+            ))
+                    .fail((exc) => {
+                        this.dialogErrorSet(_("Max Memory settings failed to change with following error: ") + exc.message);
+                    })
+                    .then(() => {
+                        if (vm.state === 'running' && this.state.memory !== vm.currentMemory) {
+                            // If machine is running, run-time config needs to be changed too
+                            dispatch(setMemory(
+                                vm,
+                                memory
+                            ))
+                                    .fail((exc) => {
+                                        this.dialogErrorSet(_("Memory settings failed to change with following error: ") + exc.message);
+                                    })
+                                    .then(() => {
+                                        dispatch(getVm({ connectionName: vm.connectionName, id: vm.id }));
```

Comment:
As we duscussed yesterday there is a signal that is generated when memory ballooning changes, so you just need to call getVm from the handler of the event, and not here.